pkg/client/cache/store.go

```diff
@@ -164,6 +164,11 @@ func (c *cache) Index(indexName string, obj interface{}) ([]interface{}, error)
 	return c.cacheStorage.Index(indexName, obj)
 }
 
+// ListIndexNames returns a list of names used by the Index
+func (c *cache) ListIndexNames(name string) []string {
+	return c.cacheStorage.ListIndexNames(name)
+}
+
```

Comment:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  tapiohush@gmail.com
    Domain eikaas.no has exceeded the max emails per hour (35/30 (116%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext3.iad.github.net ([192.30.252.194]:46930 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by cpanel30.proisp.no with esmtps (TLSv1.2:DHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.85)
	(envelope-from <noreply@github.com>)
	id 1ZEhbn-0021SB-Ic
	for robin@eikaas.no; Mon, 13 Jul 2015 19:33:43 +0200
Date: Mon, 13 Jul 2015 10:33:38 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1436808818;
	bh=0CQ7jlmhKA5ADHnb+WKEXaPcdVqq88Ygrv0CpS05R80=;
	h=From:Reply-To:To:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=GidexJiuUpY2dojh+l+lDLixoI+9eexgUgjsKXH2bmrNdnS2YYtKaCxLzktyg0/kl
	 6eoci9NzR0UEUTRoFMCc4atctTq26M7pxvLb5ObFO/DZrIW835pnezA5Qz4LWSx+zp
	 hHHloFjZXhzfWK4o4QwHIAh4tEqy3ErXey8S6HeA=
From: Mark Turansky <notifications@github.com>
Reply-To: GoogleCloudPlatform/kubernetes <reply@reply.github.com>
To: GoogleCloudPlatform/kubernetes <kubernetes@noreply.github.com>
Message-ID: <GoogleCloudPlatform/kubernetes/pull/10833/r34488498@github.com>
In-Reply-To: <GoogleCloudPlatform/kubernetes/pull/10833@github.com>
References: <GoogleCloudPlatform/kubernetes/pull/10833@github.com>
Subject: Re: [kubernetes] Allow PV/Claim matching on inexact access modes
 (#10833)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_55a3f6727ceff_413e3fc7092332b8100912c";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: markturansky
X-GitHub-Recipient: eikaas
List-ID: GoogleCloudPlatform/kubernetes
 <kubernetes.GoogleCloudPlatform.github.com>
List-Archive: https://github.com/GoogleCloudPlatform/kubernetes
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+005f5f6735bab03af0ee60bb00735a346c78a364c9d75a4d92cf0000000111bbb87292a170ce020e40b2@reply.github.com>,
 <https://github.com/notifications/unsubscribe/AF9fZzaW4aLu2rfCSbOz0JZnlU3ng8Pmks5oc-3ygaJpZM4FTmqH>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: robin@eikaas.no
X-Spam-Status: No, score=-0.9
X-Spam-Score: -8
X-Spam-Bar: /
X-Ham-Report: Spam detection software, running on the system "cpanel30.proisp.no",
 has NOT identified this incoming email as spam.  The original
 message has been attached to this so you can view it or label
 similar future email.  If you have any questions, see
 root\@localhost for details.
 
 Content preview:  > @@ -164,6 +164,11 @@ func (c *cache) Index(indexName string,
    obj interface{}) ([]interface{}, error) > return c.cacheStorage.Index(indexName,
    obj) > } > > +// ListIndexNames returns a list of names used by the Index
    > +func (c *cache) ListIndexNames(name string) []string { > + return c.cacheStorage.ListIndexNames(name)
    > +} > + [...] 
 
 Content analysis details:   (-0.9 points, 4.0 required)
 
  pts rule name              description
 ---- ---------------------- --------------------------------------------------
  0.0 URIBL_BLOCKED          ADMINISTRATOR NOTICE: The query to URIBL was blocked.
                             See
                             http://wiki.apache.org/spamassassin/DnsBlocklists#dnsbl-block
                              for more information.
                             [URIs: github.com]
 -0.0 RCVD_IN_MSPIKE_H4      RBL: Very Good reputation (+4)
                             [192.30.252.194 listed in wl.mailspike.net]
 -0.0 SPF_PASS               SPF: sender matches SPF record
 -1.9 BAYES_00               BODY: Bayes spam probability is 0 to 1%
                             [score: 0.0000]
  0.0 HTML_MESSAGE           BODY: HTML included in message
  1.1 HTML_IMAGE_ONLY_16     BODY: HTML: images with 1200-1600 bytes of words
 -0.1 DKIM_VALID_AU          Message has a valid DKIM or DK signature from author's
                             domain
 -0.1 DKIM_VALID             Message has at least one valid DKIM or DK signature
  0.1 DKIM_SIGNED            Message has a DKIM or DK signature, not necessarily valid
 -0.0 RCVD_IN_MSPIKE_WL      Mailspike good senders
X-Spam-Flag: NO


----==_mimepart_55a3f6727ceff_413e3fc7092332b8100912c
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

> @@ -164,6 +164,11 @@ func (c *cache) Index(indexName string, obj interface{}) ([]interface{}, error)
>  	return c.cacheStorage.Index(indexName, obj)
>  }
>  
> +// ListIndexNames returns a list of names used by the Index
> +func (c *cache) ListIndexNames(name string) []string {
> +	return c.cacheStorage.ListIndexNames(name)
> +}
> +

@derekwaynecarr would you mind reviewing this change?  I added the ability to retrieve index names from an Index.

---
Reply to this email directly or view it on GitHub:
https://github.com/GoogleCloudPlatform/kubernetes/pull/10833/files#r34488498
----==_mimepart_55a3f6727ceff_413e3fc7092332b8100912c
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>In <a href="https://github.com/GoogleCloudPlatform/kubernetes/pull/10833#discussion_r34488498">pkg/client/cache/store.go</a>:</p>
<pre style='color:#555'>&gt; @@ -164,6 +164,11 @@ func (c *cache) Index(indexName string, obj interface{}) ([]interface{}, error)
&gt;  	return c.cacheStorage.Index(indexName, obj)
&gt;  }
&gt;  
&gt; +// ListIndexNames returns a list of names used by the Index
&gt; +func (c *cache) ListIndexNames(name string) []string {
&gt; +	return c.cacheStorage.ListIndexNames(name)
&gt; +}
&gt; +
</pre>
<p><a href="https://github.com/derekwaynecarr" class="user-mention">@derekwaynecarr</a> would you mind reviewing this change?  I added the ability to retrieve index names from an Index.</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br>Reply to this email directly or <a href="https://github.com/GoogleCloudPlatform/kubernetes/pull/10833/files#r34488498">view it on GitHub</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/AF9fZ_mcn2LEYdgj9ZkdvxxqB8G878vAks5oc-3ygaJpZM4FTmqH.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
  <div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
    <link itemprop="url" href="https://github.com/GoogleCloudPlatform/kubernetes/pull/10833/files#r34488498"></link>
    <meta itemprop="name" content="View Pull Request"></meta>
  </div>
  <meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

----==_mimepart_55a3f6727ceff_413e3fc7092332b8100912c--